.github/workflows/scan.yml

```diff
@@ -10,14 +10,23 @@ permissions: {}
 
 jobs:
   scan:
+    strategy:
+      fail-fast: false
+      matrix:
+        branch: [ main, release-2.2, release-2.1, release-2.0 ]
     name: Trivy
     runs-on: ubuntu-latest
     steps:
       - name: Check out code
         uses: actions/checkout@v3.5.3
+        with:
+          ref: ${{ matrix.branch }}
+      - name: Calculate go version
+        id: vars
+        run: echo "go_version=$(make go-version)" >> $GITHUB_OUTPUT
       - name: Setup go
         uses: actions/setup-go@v4
         with:
-          go-version: 1.20
+          go-version: ${{ steps.vars.outputs.go_version }}
```

Comment:
Why not use go-version-file and point to the go mod file?

Reply:
```yaml
        go-version-file: '${{ github.workspace }}/go.mod'
```

Like this.

Reply:
Ah, I don't know that way. It looks simpler and has no dependency on `make go-version` support.

I just followed capi and capv pattern:
https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/scan.yml#L30
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/.github/workflows/scan.yaml#L30

Reply:
@wyike  So.. Are you going to change it? :D

Reply:
sorry for the late response. Let me try.

Reply:
It becomes quite simple.

Give a try run in my folked repo: https://github.com/wyike/cluster-api-provider-aws/actions/runs/5699290555/job/15448043792

Reply:
Nice!